feedback/lib/src/feedback_widget.dart

```diff
@@ -294,3 +306,79 @@ class FeedbackWidgetState extends State<FeedbackWidget>
     FocusScope.of(context).requestFocus(FocusNode());
   }
 }
+
+const _screenshotId = 'screenshot_id';
+const _controlsColumnId = 'controls_column_id';
+const _sheetId = 'sheet_id';
+
+class _FeedbackLayoutDelegate extends MultiChildLayoutDelegate {
```

Comment:
Do you mind explaining why this is needed? I'm not really familiar with `CustomMultiChildLayout` and `MultiChildLayoutDelegate`.

Reply:
`CustomMultiChildLayout` is a convenience widget over the underlying Flutter layout system (I think? I'm not actually an expert on it). So when you use a `Stack` or `Column` etc.,   it's using something pretty similar under the hood. It's a lot simpler than it looks at first glance.

It has an arbitrary number of children that are all passed in with an ID. The delegate interacts with the children using their IDs..
You call `layoutChild(someId, constraints)` to tell a given child widget to build itself within the given box constraints. This is like wrapping a `ConstrainedBox` around the child.
You call `positionChild(...)` to position it. This is like wrapping `Positioned` around it in a stack.
The key advantage is that you choose the order in which to lay the children out and `layoutChild` gives you the size the child ended up taking up. This way, children you layout later can have arbitrary dependencies on the sizes of children you laid out earlier.

The biggest reason I used it over just a stack is that the screenshot widget's size and position are dependent on the width of the controls column (if we want the layout to be properly reactive).
To achieve this inter-dependency with a stack, I'd either have to (AFAIK):

1. Use `GlobalContext` and a post frame callback to get the width after the first draw
2. Force the controls column to be a preset width

Both felt sloppy and are known anti patterns.

Reply:
Thanks!